assets/blocks/core-image/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * External dependencies
+ */
+import { addFilter } from '@wordpress/hooks';
+import { useSelect, useDispatch } from '@wordpress/data';
+import { Spinner } from '@wordpress/components';
+import { createHigherOrderComponent } from '@wordpress/compose';
+import './style.scss';
+
+/**
+ * Add image credit meta to core/image block attributes
+ */
+addFilter(
+	'blocks.registerBlockType',
+	'newspack-plugin/register-hook/core-image',
+	( settings, name ) => {
+		if ( name !== 'core/image' ) {
+			return settings;
+		}
+		return {
+			...settings,
+			attributes: {
+				...settings.attributes,
+				meta: {
+					type: 'object',
+					default: {},
+				},
+			},
+		};
+	}
+);
+
+/**
+ * Display spinner and lock post saving until meta attributes are added to block
+ */
+const AttributesLoader = props => {
+	const imageId = props.attributes.id;
+	const { lockPostSaving, unlockPostSaving } = useDispatch( 'core/editor' );
+	const { meta } = useSelect( select => select( 'core' ).getMedia( imageId ), [ imageId ] ) ?? {
+		meta: {},
+	};
+	// Meta added, proceed
+	if ( Object.keys( meta ).length ) {
+		props.setAttributes( { meta } );
+		unlockPostSaving( 'attachment-meta-empty' );
+	} else {
+		lockPostSaving( 'attachment-meta-empty' );
+	}
+
+	return (
+		<>
+			{ ! Object.keys( meta ).length && (
+				<div className="newspack-block__core-image-background">
+					<div className="newspack-block__core-image-spinner">
+						<Spinner />
+					</div>
+				</div>
+			) }
+		</>
+	);
+};
+
+/**
+ * Populate attributes with meta data.
+ */
+addFilter(
+	'editor.BlockEdit',
+	'newspack-plugin/block-edit-hook/core-image',
+	createHigherOrderComponent( BlockEdit => {
+		const blockEditComponent = props => {
+			if ( props.name === 'core/image' ) {
+				return (
+					<div className="newspack-block__core-image">
+						<BlockEdit { ...props } />
+						<AttributesLoader { ...props } />
```

Comment:
Suggestion: render the credits here, so it's a more WYSIWYG experience.

Reply:
Nice suggestion, this did cross my mind, I decided against adding any net-new functionality to try stay within scope. However, I've come up with a simple way of doing this without significant code additions.

See - [feat(blocks): make image credit display in editor](https://github.com/Automattic/newspack-plugin/pull/2936/commits/d4b9138c27875f62c936e3d2e940c27a47b7bf71)

![Screenshot 2024-02-19 at 10 21 23](https://github.com/Automattic/newspack-plugin/assets/3676975/bfbea050-789c-46d4-a9c7-511b6dbd79ea)

Look forward to your feedback.